lib/rubocop/cop/rspec/empty_example_group.rb

```diff
@@ -68,13 +68,20 @@ class EmptyExampleGroup < Cop
         PATTERN
 
         def on_block(node)
-          return unless example_group?(node) && !contains_example?(node)
+          return unless example_group?(node)
+          return if node.children.any? do |c|
+            c && should_contain_example?(c) && contains_example?(c)
+          end
 
           add_offense(node.send_node)
         end
 
         private
 
+        def should_contain_example?(node)
+          !hook?(node) && !let?(node) && !subject?(node)
```

Comment:
Instead of listing possible node types that may have examples in them, what about replacing the recursive search with check only on immediate children and passing on an example or example group (as @pirj suggested in the original issue)

Reply:
Yep, that's the whole point, I've stepped into this so many times now - `def_node_search` is recursive, while `def_node_matcher` only matches direct children.

Reply:
This was the first thing I tried, but it caused a number of test failures. Pushing for reference